core/logger/Logger.cpp

```diff
@@ -75,6 +75,25 @@ static std::string MapLevelToString(level::level_enum level) {
     }
 }
 
+std::string toUpperCase(const std::string& str) {
+    std::string result = str;
+    for (char& c : result) {
+        c = std::toupper(c);
+    }
+    return result;
+}
+
+bool ReadLogLevelFromEnv(level::level_enum* logLevel, std::string& aliyun_logtail_log_level) {
```

Comment:
Pass by reference if you want to modify the value. Pass by const reference or value  if you don't.